### PR TITLE
Add markdown tutorial to docs

### PR DESCRIPTION
**Pull request by marload**
Added Tutorial 1 and 2 to docs. I checked that markdown looks normal. However, the results shown on the github page can be checked after the merge.

Thanks!

### CODE REVIEW

File: docs/templates/tutorials/01-basic-tutorial.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
+# 01. Basic Tutorial
+
+In Basic Tutorial, you can learn the following things:
```
Comment (marctorsoc) [inline docs/templates/tutorials/01-basic-tutorial.md:3]: In this tutorial, you can learn how to:
* Define a Search Space
* Define and optimize an Objective Function

File: docs/templates/tutorials/01-basic-tutorial.md [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+fig = plt.figure()
+plt.plot(x, y)
+plt.show()
+```
+
+
+![png](01-basic-tutorial_files/01-basic-tutorial_6_0.png)
+
+
+We are trying to optimize the `objective` by changing the HyperParameter $x$. That's why we will declare a search space for $x$. The functions related to the search space are implemented in `hyperopt.hp`. The list is as follows.
```
Comment (marctorsoc) [inline docs/templates/tutorials/01-basic-tutorial.md:44]: I would change

`objective` - > objective function (without `)

File: docs/templates/tutorials/01-basic-tutorial.md [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+* `hp.qnormal(label, mu, sigma, q)`
+* `hp.qlognormal(label, mu, sigma, q)`
+* `hp.choice(label, list)`
+
+We will use the most basic `hp.uniform` in this tutorial.
+    
+
+
+```python
+# Define the search space of x between -10 and 10.
```
Comment (marctorsoc) [inline docs/templates/tutorials/01-basic-tutorial.md:62]: Maybe a comment to say that uniform means that all continuous values have the same probability to be sampled

File: docs/templates/tutorials/01-basic-tutorial.md [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+```
+
+Now, there's only one last step left. So far, we have defined a function of purpose, and we have defined a search space for $x$. Now we can search through the search space $x$ and find the value of $x$ that can optimize the `objective`. HyperOpt performs it using `fmin`.
+
+
+```python
+best = fmin(
+    fn=objective, # Objective Function to optimize
+    space=space, # Hyperparameter's Search Space
+    algo=tpe.suggest, # Optimization algorithm (representative TPE)
```
Comment (marctorsoc) [inline docs/templates/tutorials/01-basic-tutorial.md:73]: What do mean by representative?

File: docs/templates/tutorials/01-basic-tutorial.md [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+We will use the most basic `hp.uniform` in this tutorial.
+    
+
+
+```python
+# Define the search space of x between -10 and 10.
+space = hp.uniform('x', -10, 10)
+```
+
+Now, there's only one last step left. So far, we have defined a function of purpose, and we have defined a search space for $x$. Now we can search through the search space $x$ and find the value of $x$ that can optimize the `objective`. HyperOpt performs it using `fmin`.
```
Comment (marctorsoc) [inline docs/templates/tutorials/01-basic-tutorial.md:66]: I think "objective function" is more popular.

What about something like 

```
So far, we have defined an objective function and a search space for $x$. Now we can search on the search space
```

File: docs/templates/tutorials/02-multiple-parameter-tutorial.md [hunk 6]
```diff
@@ -0,0 +1,126 @@
+# 02. MultipleParameterTutorial
+
+In [01.BasicTutorial](https://github.com/hyperopt/hyperopt/blob/master/tutorial/01.BasicTutorial.ipynb), you learned about optimizing a single HyperParameter function. At MultipleParameterTurial, you learn the following things.
+
+* Optimize the Objective Function with Multiple HyperParameters
+* Define various search space
```
[the remark this answers is not in the document]
Comment (marctorsoc) [reply docs/templates/tutorials/02-multiple-parameter-tutorial.md:6]: In this tutoria, you will learn how to
* ...
* Define various search spaces

File: docs/templates/tutorials/02-multiple-parameter-tutorial.md [hunk 7]
```diff
@@ -0,0 +1,126 @@
...
+
+
+```python
+# Import HyperOpt Library
+from hyperopt import tpe, hp, fmin
+import numpy as np
+```
+
+Declares a purpose function to optimize. Unlike last time, we will optimize the function with two Hyperparameters, $x$ and $y$.
```
Comment (marctorsoc) [inline docs/templates/tutorials/02-multiple-parameter-tutorial.md:15]: I think better to use the term "objective function"

File: docs/templates/tutorials/02-multiple-parameter-tutorial.md [hunk 8]
```diff
@@ -0,0 +1,126 @@
...
+    max_evals=1000 # Number of optimization attempts
+)
+print(best)
+```
+
+    100%|██████████| 1000/1000 [00:08<00:00, 124.16trial/s, best loss: -0.9999998753676712]
+    {'x': 4.283678790265057, 'y': 1.9626514514966948}
+
+
+## Using various search spaces
```
Comment (marctorsoc) [inline docs/templates/tutorials/02-multiple-parameter-tutorial.md:82]: this section seems a bit outlier. It's not about multiple parameters. What if you add some more examples of search spaces with multiple parameters but using other distributions? (for the same objective function). Using `f: x --> -x` is not very meaningful

**Comment by marload** [comment]
@marctorsoc 
My native language is not English, so there are many awkward parts in the sentence. I'll change it as you suggested. Thank you.

**Comment by marctorsoc** [comment]
> @marctorsoc
> My native language is not English, so there are many awkward parts in the sentence. I'll change it as you suggested. Thank you.

sure, no worries. Mine is not English either. I just learned after many people (like me now) correcting me :D

**Comment by maxpumperla** [comment]
@marload neither of us is a native speaker ;)  Just please add Marc' suggestions and we're good to go.

**Comment by marload** [comment]
@marctorsoc @maxpumperla 

The proposal has been applied.
By any chance, how about creating a channel for HyperOpt members to communicate with each other?

**Comment by marctorsoc** [comment]
> @marctorsoc @maxpumperla
> 
> The proposal has been applied.
> By any chance, how about creating a channel for HyperOpt members to communicate with each other?

I like this idea. https://hey.space/pricing/ seems a good option (<10 team members for the free option), but any platform sounds good to me

**Comment by marctorsoc** [comment]
btw, the tests failed @marload . I'll remove the approval until it's fixed

**Comment by marload** [comment]
@marctorsoc 
To be honest, I don't know why that mongodb_exp error occurs. That error occurs randomly. First, I'll run the test again.

**Comment by marload** [comment]
@marctorsoc
There is definitely a problem with `mongodb_exp_test`. I only restart build without any code modification. But the test passed.
I'll look into the error of the test.
Thanks

PS) I also like the platform you mentioned. @maxpumperla  What do you think?

**Pull request merged**